apps/calendar/js/views/modify_event.js

```diff
@@ -166,128 +159,34 @@ ModifyEvent.prototype = {
     // really rare occasions "onfirstseen" is called after the EventBase
     // removed the "loading" class from the root element (seen it happen less
     // than 1% of the time)
-    this.getEl('calendarId').classList.add(self.LOADING);
-
-    var calendarStore = core.storeFactory.get('Calendar');
-    calendarStore.all(function(err, calendars) {
-      if (err) {
-        return console.error('Could not build list of calendars');
-      }
-
-      var pending = 0;
-      var self = this;
-
-      function next() {
-        if (!--pending) {
-          self.getEl('calendarId').classList.remove(self.LOADING);
-
-          if (self.onafteronfirstseen) {
-            self.onafteronfirstseen();
-          }
-        }
-      }
-
-      for (var id in calendars) {
-        pending++;
-        this._addCalendarId(id, calendars[id], next);
-      }
-
-    }.bind(this));
+    this.getEl('calendarId').classList.add(this.LOADING);
   },
 
-  /**
-   * Updates a calendar id option.
-   *
-   * @param {String} id calendar id.
-   * @param {Calendar.Model.Calendar} calendar model.
-   */
-  _updateCalendarId: function(id, calendar) {
-    var element = this.getEl('calendarId');
-    var option = element.querySelector('[value="' + id + '"]');
-    var store = core.storeFactory.get('Calendar');
-
-    store.providerFor(calendar, function(err, provider) {
-      var caps = provider.calendarCapabilities(
-        calendar
-      );
-
-      if (!caps.canCreateEvent) {
-        this._removeCalendarId(id);
-        return;
-      }
+  _renderCalendars: function(records) {
+    // remove old calendars (this will be called every time we have a change on
+    // any calendar)
+    var select = this.getEl('calendarId');
+    // remove all the options
+    select.options.length = 0;
+    select.classList.remove(this.LOADING);
 
-      if (option) {
-        option.text = calendar.remote.name;
-      }
+    records = records.filter(r => r.capabilities.canCreateEvent);
```

Comment:
random style thing: what do you think about chaining the call to `filter` and the call to `forEach`?